Source/x11/XGServerWindow.m

```diff
@@ -4509,13 +4512,33 @@ - (NSSize) resolutionForScreen: (int)screen_num
 
 - (NSRect) boundsForScreen: (int)screen
 {
- if (screen < 0 || screen >= ScreenCount(dpy))
-   {
-     NSLog(@"Invalidparam: no screen %d", screen);
-     return NSZeroRect;
-   }
- return NSMakeRect(0, 0, DisplayWidth(dpy, screen),
-		   DisplayHeight(dpy, screen));
+  NSRect boundsRect = {{0,0},{0,0}};
+  
+  if (screen < 0 || screen >= ScreenCount(dpy))
+    {
+      NSLog(@"Invalidparam: no screen %d", screen);
+      return NSZeroRect;
+    }
+  
+#ifdef HAVE_XRANDR
+  XRRScreenResources *screen_res;
```

Comment:
As this operations look expensive, should we cache this information in NSScreen resetScreens and reuse the cached values?

Reply:
Interesting idea, thanks. I'll do it.